internal/pkg/service/stream/encryption/config.go

```diff
@@ -0,0 +1,52 @@
+package encryption
+
+type Config struct {
+	Provider Provider      `json:"provider" configKey:"provider" validate:"required,oneof=native gcp aws azure" configUsage:"Encryption provider."`
+	Native   *NativeConfig `json:"native" configKey:"native" validate:"required_if=Provider native"`
+	GCP      *GCPConfig    `json:"gcp" configKey:"gcp" validate:"required_if=Provider gcp"`
+	AWS      *AWSConfig    `json:"aws" configKey:"aws" validate:"required_if=Provider aws"`
+	Azure    *AzureConfig  `json:"azure" configKey:"azure" validate:"required_if=Provider azure"`
+}
+
+type NativeConfig struct {
+	SecretKey string `json:"secretKey" configKey:"secretKey" sensitive:"true" validate:"required,len=32" configUsage:"Secret key for local encryption. Do not use in production."`
+}
+
+type GCPConfig struct {
+	KMSKeyID string `json:"kmsKeyId" configKey:"kmsKeyId" validate:"required" configUsage:"Key ID in Google Cloud Key Management Service."`
+}
+
+type AWSConfig struct {
+	Region   string `json:"region" configKey:"region" validate:"required" configUsage:"AWS Region."`
+	KMSKeyID string `json:"kmsKeyId" configKey:"kmsKeyId" validate:"required" configUsage:"Key ID in AWS Key Management Service."`
+}
+
+type AzureConfig struct {
+	KeyVaultURL string `json:"keyVaultUrl" configKey:"keyVaultUrl" validate:"required,url" configUsage:"Azure Key Vault URL."`
+	KeyName     string `json:"keyName" configKey:"keyName" validate:"required" configUsage:"Key name in the vault."`
+}
+
+func NewConfig() Config {
+	return Config{
+		Provider: ProviderNative,
+		Native:   &NativeConfig{},
+		GCP:      &GCPConfig{},
+		AWS:      &AWSConfig{},
+		Azure:    &AzureConfig{},
+	}
+}
+
+func (c *Config) Normalize() {
```

Comment:
Have you check that?

Reply:
You mean where this method is called? It's part of `ValueWithNormalization` interface and invoked here:

https://github.com/keboola/keboola-as-code/blob/60c9f4f79f093a2a722d933c80028109d0b4a662/internal/pkg/service/common/configmap/bind.go#L245-L248